src/codeu/chat/server/Server.java

```diff
@@ -283,4 +299,23 @@ public void run() {
       }
     };
   }
-}
+
+  public Model deserialize(File filepath){
+    try {
+      FileInputStream fs = new FileInputStream(filepath); // what else do I add here
+      ObjectInputStream os = new ObjectInputStream(fs);
+
+      // saving the objects that are serialized
+      // Is there only one object (the data) that needs to be serialized?
+      Object one = os.readObject();
+
+      // Cast the object back to Model
+      Model data = (Model) one;
+      return data;
+    } catch (IOException i) {
+      i.printStackTrace();
+    } catch (ClassNotFoundException c) {
+      c.printStackTrace();
+    }
+    return null;
+  }
```

Comment:
You get a compilation error if you run `python build.py build`, you're missing a final `}`